core/plannernotes.c

```diff
@@ -193,6 +196,8 @@ void add_plan_to_notes(struct diveplan *diveplan, struct dive *dive, bool show_d
 		    !rebreatherchange_before &&
 		    !rebreatherchange_after)
 			continue;
+		if (lastdepth == 0 && dp->depth.mm == 0 && !dp->next)
+			continue;
```

Comment:
so that's the end of the plan - why do we skip the rest of the loop? A comment here might be helpful when we look at this code in a few years

Reply:
We skip the rest as the surface segment should not show up in the notes. Will add a comment.